pkg/inject/report.go

```diff
@@ -158,11 +158,11 @@ func (r *Report) disableByAnnotation(conf *ResourceConfig) (bool, string, string
 		return podAnnotation == k8s.ProxyInjectDisabled, "", ""
 	}
 
-	if !isInjectAnnotationValid(conf.nsAnnotations) {
+	if !(nsAnnotation == k8s.ProxyInjectEnabled || nsAnnotation == k8s.ProxyInjectDisabled) {
 		return true, invalidInjectAnnotationNamespace, ""
 	}
 
-	if !isInjectAnnotationValid(conf.pod.meta.Annotations) {
+	if !(podAnnotation == k8s.ProxyInjectEnabled || podAnnotation == k8s.ProxyInjectDisabled) {
```

Comment:
Seems like your test is failing because before the assumption was that not gaing this annotation set is ok. Now you treat it like an error. The reason for this is because if you do a lookup on a map and you assign it to a value, if your map is string -> string and you have a missing key, yur result will be an empty string. You need to add a check for that to your test condition. Namely: 
`!(podAnnotation == k8s.ProxyInjectEnabled || podAnnotation == k8s.ProxyInjectDisabled) && podAnnotation != ""`

Same for `nsAnnotation`. Hope this helps.

Reply:
@zaharidichev that's what we began with! @ihcsim suggested that we remove the `podAnnotation != ""` check (https://github.com/linkerd/linkerd2/pull/3933#discussion_r367562004). Seems like we'd have to stick to having that check then

Reply:
Thats what it seems to be. Hope I am not overlooking something, @ihcsim ?

Reply:
Ah yes... the `podAnnotation !=""` and `nsAnnotation != ""` are needed after all, because empty string values are valid for these annotations. Thanks for pointing that out.